File: map2d/src/test/java/net/ponderingprogrammer/map2d/ChebyshevFixedSquareMap2dTest.java
package net.ponderingprogrammer.map2d;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

class ChebyshevFixedSquareMap2dTest {

    @Test
    void findAdjacentCells() {
        var map = new ChebyshevFixedSquareMap2d<Integer>(9, 9);

        Assertions.assertEquals(8, map.findAdjacentCells(1, 1).size());
        Assertions.assertEquals(3, map.findAdjacentCells(0, 0).size());
        Assertions.assertEquals(5, map.findAdjacentCells(1, 0).size());
    }

    @Test
    void toGraph() {
        var map = new ChebyshevFixedSquareMap2d<Integer>(2, 2);
        var graph = map.toCellGraph();

        Assertions.assertEquals(4, graph.getNodes().size());
        Assertions.assertEquals(6, graph.getEdges().size());
    }

}

File: projects/dotnet/NProcGen/tests/PonderingProgrammer.Map2d.Tests/ChebyshevFixedSquareMap2dTest.cs
using System.Linq;
using Xunit;

namespace PonderingProgrammer.Map2d.Tests
{
    public class ChebyshevFixedSquareMap2dTest
    {
        [Fact]
        public void FindAdjacentCells()
        {
            var map = new ChebyshevFixedSquareMap2d<bool>(9, 9);

            Assert.Equal(8, map.FindAdjacentCells(1, 1).Count());
            Assert.Equal(3, map.FindAdjacentCells(0, 0).Count());
            Assert.Equal(5, map.FindAdjacentCells(1, 0).Count());
        }

        [Fact]
        public void ToGraph()
        {
            var map = new ChebyshevFixedSquareMap2d<bool>(2, 2);
            var graph = map.ToCellGraph();

            Assert.Equal(4, graph.Nodes.Count);
            Assert.Equal(6, graph.Edges.Count);
        }
    }
}
